In [6]:
pip install kaggle

Note: you may need to restart the kernel to use updated packages.


#### Setup Directory Structure

In [9]:
data_folder = os.path.join(os.getcwd(), 'data')
#Create the data directory
os.makedirs(data_folder, exist_ok=True)

kaggle_folder = os.path.join(os.getcwd(), '.kaggle')
#Create the data directory
os.makedirs(kaggle_folder, exist_ok=True)

#Manually upload the kaggle.json generated from kaggle in .kaggle folder which conains the username and the key

In [10]:
kaggle_key_folder = '/home/azureuser/.kaggle'
#Create the .kaggle key directory
os.makedirs(kaggle_key_folder, exist_ok=True)

In [23]:
import shutil
import os
kaggle_file = kaggle_folder + '/kaggle.json'
shutil.copy(kaggle_file, kaggle_key_folder)
os.remove(kaggle_file)

In [12]:
#No other users can read
!chmod 600 /home/azureuser/.kaggle/kaggle.json

In [13]:
import kaggle

In [14]:
!kaggle --version

Kaggle API 1.5.12


In [15]:
#List the competition files
!kaggle competitions files -c zillow-prize-1

name                          size  creationDate         
---------------------------  -----  -------------------  
sample_submission.csv         60MB  2019-12-11 04:25:25  
train_2017.csv                 3MB  2019-12-11 04:25:25  
properties_2017.csv          620MB  2019-12-11 04:25:25  
properties_2016.csv          619MB  2019-12-11 04:25:25  
train_2016_v2.csv              2MB  2019-12-11 04:25:25  
zillow_data_dictionary.xlsx   19KB  2019-12-11 04:25:25  


In [16]:
#Download the zipped file at the current working directory
!kaggle competitions download -c zillow-prize-1

403 - Forbidden


#### Have to accept the terms and condition else will get "403 - Forbidden"

In [17]:
#Download the zipped file at the current working directory
#Lets try again
!kaggle competitions download -c zillow-prize-1

 95%|█████████████████████████████████████▉  | 322M/340M [00:05<00:00, 66.7MB/s]
100%|████████████████████████████████████████| 340M/340M [00:08<00:00, 42.4MB/s]


In [19]:
#Extract files in the data folder
import zipfile
with zipfile.ZipFile("zillow-prize-1.zip","r") as zip_ref:
    zip_ref.extractall(data_folder)

In [20]:
#List the folder structure and the files
for root, directories, files in os.walk(data_folder, topdown=True):
	for name in files:
		print(os.path.join(root, name))

/mnt/batch/tasks/shared/LS_root/mounts/clusters/compute-cpu-ds12-v2/code/Users/rabiswas/HomeValuePrediction/data/properties_2016.csv
/mnt/batch/tasks/shared/LS_root/mounts/clusters/compute-cpu-ds12-v2/code/Users/rabiswas/HomeValuePrediction/data/properties_2017.csv
/mnt/batch/tasks/shared/LS_root/mounts/clusters/compute-cpu-ds12-v2/code/Users/rabiswas/HomeValuePrediction/data/sample_submission.csv
/mnt/batch/tasks/shared/LS_root/mounts/clusters/compute-cpu-ds12-v2/code/Users/rabiswas/HomeValuePrediction/data/train_2016_v2.csv
/mnt/batch/tasks/shared/LS_root/mounts/clusters/compute-cpu-ds12-v2/code/Users/rabiswas/HomeValuePrediction/data/train_2017.csv
/mnt/batch/tasks/shared/LS_root/mounts/clusters/compute-cpu-ds12-v2/code/Users/rabiswas/HomeValuePrediction/data/zillow_data_dictionary.xlsx


In [21]:
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

In [22]:
import pandas as pd
properties_file = '/mnt/batch/tasks/shared/LS_root/mounts/clusters/compute-cpu-ds12-v2/code/Users/rabiswas/HomeValuePrediction/data/properties_2017.csv'
df_properties = pd.read_csv(properties_file)
#It is easier to view the data if we transpose
df_properties.head(3).transpose()

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3072: DtypeWarning: Columns (49) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


0            1            2
parcelid                         10754147     10759547     10843547
airconditioningtypeid                 NaN          NaN          NaN
architecturalstyletypeid              NaN          NaN          NaN
basementsqft                          NaN          NaN          NaN
bathroomcnt                             0            0            0
bedroomcnt                              0            0            0
buildingclasstypeid                   NaN          NaN            5
buildingqualitytypeid                 NaN          NaN          NaN
calculatedbathnbr                     NaN          NaN          NaN
decktypeid                            NaN          NaN          NaN
finishedfloor1squarefeet              NaN          NaN          NaN
calculatedfinishedsquarefeet          NaN          NaN        73026
finishedsquarefeet12                  NaN          NaN          NaN
finishedsquarefeet13                  NaN          NaN          NaN
finishedsquarefeet15                  NaN          NaN        73026
finishedsquarefeet50                  NaN          NaN          NaN
finishedsquarefeet6                   NaN          NaN          NaN
fips                                 6037         6037         6037
fireplacecnt                          NaN          NaN          NaN
fullbathcnt                           NaN          NaN          NaN
garagecarcnt                          NaN          NaN          NaN
garagetotalsqft                       NaN          NaN          NaN
hashottuborspa                        NaN          NaN          NaN
heatingorsystemtypeid                 NaN          NaN          NaN
latitude                      3.41444e+07  3.41404e+07  3.39894e+07
longitude                    -1.18654e+08 -1.18625e+08 -1.18395e+08
lotsizesquarefeet                   85768         4083        63085
poolcnt                               NaN          NaN          NaN
poolsizesum                           NaN          NaN          NaN
pooltypeid10                          NaN          NaN          NaN
pooltypeid2                           NaN          NaN          NaN
pooltypeid7                           NaN          NaN          NaN
propertycountylandusecode            010D         0109         1200
propertylandusetypeid                 269          261           47
propertyzoningdesc                    NaN       LCA11*         LAC2
rawcensustractandblock         6.0378e+07   6.0378e+07   6.0377e+07
regionidcity                        37688        37688        51617
regionidcounty                       3101         3101         3101
regionidneighborhood                  NaN          NaN          NaN
regionidzip                         96337        96337        96095
roomcnt                                 0            0            0
storytypeid                           NaN          NaN          NaN
threequarterbathnbr                   NaN          NaN          NaN
typeconstructiontypeid                NaN          NaN          NaN
unitcnt                               NaN          NaN            2
yardbuildingsqft17                    NaN          NaN          NaN
yardbuildingsqft26                    NaN          NaN          NaN
yearbuilt                             NaN          NaN         1959
numberofstories                       NaN          NaN            1
fireplaceflag                         NaN          NaN          NaN
structuretaxvaluedollarcnt            NaN          NaN       660680
taxvaluedollarcnt                       9        27516  1.43494e+06
assessmentyear                       2016         2015         2016
landtaxvaluedollarcnt                   9        27516       774261
taxamount                             NaN          NaN      20800.4
taxdelinquencyflag                    NaN          NaN          NaN
taxdelinquencyyear                    NaN          NaN          NaN
censustractandblock                   NaN          NaN          NaN